src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -300,6 +322,10 @@ public class AddOrUpdateCompareExchangeCommand : CompareExchangeCommandBase
 
         internal const int MaxNumberOfCompareExchangeKeyBytes = 512;
         public long? ExpirationTicks;
+        
+        //Should be use only for atomic guard.
```

Comment:
why?

Reply:
The main reason to ignore compare exchange concurrency validation
is for avoiding situations where the document was deleted because it was expired
but its atomic guard wasn't deleted yet.
That lead to a situation where the client will try to get the document and will find there is no document
and will fail to create it again.

Adding this functionality to the regular compare exchange is easy to do
but will require future attention (regular maintenance) and without significant benefit.